### ISSUE TITLE
Command "${PODS_ROOT}/SwiftLint/swiftlint" not working

### ISSUE THREAD

**Issue by sphairo** [realm/SwiftLint#2829]
SwiftLint/swiftlint: No such file or directory
Command PhaseScriptExecution failed with a nonzero exit code

**Comment by dmillerconsulting** [comment]
@sphairo, did you ever find a solution to this? I am having the same problem

**Comment by a-25** [comment]
@sphairo, @dmillerconsulting: could you provide more detail?
- are you using mac os?
- on your mac, open Terminal and type commands:
```
cd <project_folder>
ls -la Pods/SwiftLint/
```
and report the output here
- type commands:
```
cd <project_folder>
Pods/SwiftLint/swiftlint
```
and report the output here
- try another script on Build Phases (if it fails there):
```
if which swiftlint >/dev/null; then
    swiftlint
else
    echo "warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint"
fi
```

It would be great if you attach the project where it fails.

**Comment by marcelofabri** [comment]
Closing this due to lack of information. Please use the issue template when filling issues.

**Comment by menjoo** [comment]
I think what was meant is that the script on Build phases referred to in the readme:
```bash
if which swiftlint >/dev/null; then
    swiftlint
else
    echo "warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint"
fi
```
Results in `Command PhaseScriptExecution failed with a nonzero exit code` when SwiftLint is installed via CocoaPods. The readme notes to call it then via `${PODS_ROOT}/SwiftLint/swiftlint`, but I is not that clear what to do with it.

When changing the script to this it works:
```bash
if which "${PODS_ROOT}/SwiftLint/swiftlint" >/dev/null; then
    ${PODS_ROOT}/SwiftLint/swiftlint
else
    echo "warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint"
fi
```
And if you also want autocorrect to run, replace the 2nd line with this
```bash
${PODS_ROOT}/SwiftLint/swiftlint autocorrect && ${PODS_ROOT}/SwiftLint/swiftlint
```

**Comment by zanyfly** [comment]
@menjoo SwiftLint doesn't work in my Xcode 12.2, I have copied the script .

**Comment by NikKovIos** [comment]
@menjoo that should be in readme. 
I've updated script:

```ruby
lint="${PODS_ROOT}/SwiftLint/swiftlint"
if which $lint >/dev/null; then
    $lint --fix && $lint
else
    echo "warning: SwiftLint not installed in cocoapods"
    exit 1
fi
```

**Comment by emrecanozturk** [comment]
It looks like there's an issue with the path to your SwiftLint script. The error indicates that the script is being treated as a directory. The path might be incorrect or improperly formatted.

In your script, try wrapping the path in double quotes to ensure that it's treated as a single path, especially if there are spaces in the path:

`"${PODS_ROOT}/SwiftLint/swiftlint"`

Make sure there are no extra characters or spaces in the path. Also, confirm that the SwiftLint script is actually located at the specified path.

If you're still encountering issues, you might want to check the value of **${PODS_ROOT}** to ensure it's pointing to the correct location. You can add an echo statement to print the value:

```
echo "PODS_ROOT: ${PODS_ROOT}"
"${PODS_ROOT}/SwiftLint/swiftlint"

```

This will print the value of **${PODS_ROOT}** to the console before executing the SwiftLint script.

Additionally, make sure that the script file itself has the correct permissions to be executed. You can use the **chmod** command to ensure that the script is executable:

`chmod +x "${PODS_ROOT}/SwiftLint/swiftlint"`

This command makes the script executable. After making these adjustments, try running your script again.
